src/charm.py

```diff
@@ -1,104 +1,432 @@
+                "bypassPaths": ["/api/info"]
+            },
+            "gitLab": {
+                "newProjectVisibility": gitlab_project_visibility,
+                "projectTag": gitlab_project_tag,
+                "uat": {
+                    "server": {
+                        # NOTE(aznashwan): these will need configuring when we
+                        # add support for relating to a Juju-managed Gitlab:
+                        "scheme": "https",
+                        "host": "gitlab.com",
+                    },
+                    "app": {
+                        # TODO(aznashwan): set these on Gitlab relation:
+                        "id": gitlab_client_id,
+                        "secret": gitlab_client_secret,
+                        "redirectURI": (
+                            "http://localhost:7070/api/auth/callback")
+                    }
+                },
+            },
+            "projectStructure": {
+                "extensionProvider": {
+                    "org.finos.legend.sdlc.server.gitlab.finos."
+                    "FinosGitlabProjectStructureExtensionProvider": {}
+                }
+            },
+            "logging": {
+                "level": server_logging_level,
+                "appenders": [{
+                    "type": "console",
+                    "logFormat": server_logging_format,
+                }]
+            },
+            "swagger": {
+                "title": "Legend SDLC",
+                "resourcePackage": "org.finos.legend.sdlc.server.resources",
+                "version": "local-snapshot",
+                "schemes": []
+            }
+        })
+
+        return None
+
+    def _update_sdlc_service_config(
+            self, container: model.Container, config: dict) -> None:
+        """Generates YAML config for the SDLC server and adds it into the
+        container via Pebble files API.
         """
-        current = self.config["thing"]
-        if current not in self._stored.things:
-            logger.debug("found a new thing: %r", current)
-            self._stored.things.append(current)
+        logger.debug(
+            "Adding following config under '%s' in container: %s",
+            SDLC_CONFIG_FILE_CONTAINER_LOCAL_PATH, config)
+        container.push(
+            SDLC_CONFIG_FILE_CONTAINER_LOCAL_PATH,
+            yaml.dump(config),
+            make_dirs=True)
+        logger.info(
+            "Successfully wrote config file in container under '%s'",
+            SDLC_CONFIG_FILE_CONTAINER_LOCAL_PATH)
 
-    def _on_fortune_action(self, event):
-        """Just an example to show how to receive actions.
+    def _restart_sdlc_service(self, container: model.Container) -> None:
+        """Restarts the SDLC service using the Pebble container API.
+        """
+        logger.debug("Restarting SDLC service")
+        container.restart("sdlc")
+        logger.debug("Successfully issues SDLC service restart")
```

Comment:
This method seems a little pointless to me -- only called once and wraps a trivial one-liner